pkg/perf/dump.go

```diff
@@ -83,21 +87,53 @@ func ReadJITdump(logger log.Logger, fileName string) (Map, error) {
 	return Map{Path: fileName, addrs: addrs, stringTable: st}, nil
 }
 
-func NewJITDumpCache(logger log.Logger, reg prometheus.Registerer, profilingDuration time.Duration) *JITDumpCache {
-	return &JITDumpCache{
+func NewJITDumpCache(
+	logger log.Logger,
+	reg prometheus.Registerer,
+	tmpDir string,
+	profilingDuration time.Duration,
+) *JITDumpCache {
+	c := &JITDumpCache{
 		logger: logger,
-		cache: cache.NewLRUCacheWithTTL[string, jitdumpCacheValue](
-			prometheus.WrapRegistererWith(prometheus.Labels{"cache": "jitdump_cache"}, reg),
-			512,
-			10*profilingDuration,
-		),
+		tmpDir: tmpDir,
+	}
+
+	f := func(key string, value jitdumpCacheValue) {
+		if err := value.f.Close(); err != nil {
+			level.Error(logger).Log("msg", "failed to close perf map file", "err", err)
+		}
+
+		if err := os.Remove(key); err != nil {
+			level.Error(logger).Log("msg", "failed to remove perf map file", "err", err)
+		}
 	}
+
+	c.cache = cache.NewLRUCacheWithEvictionTTL[string, jitdumpCacheValue](
+		prometheus.WrapRegistererWith(prometheus.Labels{"cache": "jitdump_cache"}, reg),
+		512,
+		10*profilingDuration,
+		f,
+	)
+
+	return c
+}
+
+func (p *JITDumpCache) path(pid int, fileName string) string {
+	return p.pathForKey(key(pid, fileName))
+}
+
+func (p *JITDumpCache) pathForKey(key string) string {
+	return filepath.Join(p.tmpDir, key)
+}
+
+func key(pid int, fileName string) string {
+	return fmt.Sprintf("/proc/%d/root%s", pid, fileName)
```

Comment:
Just in case the filename doesn't contain a trailing path, can we use `filepath.Join`?